evm_arithmetization/src/generation/jumpdest.rs

```diff
@@ -0,0 +1,191 @@
+use std::cmp::max;
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    ops::{Deref, DerefMut},
+};
+
+use itertools::{sorted, Itertools};
+use keccak_hash::H256;
+use serde::{Deserialize, Serialize};
+
+/// Each `CodeAddress` can be called one or more times, each time creating a new
+/// `Context`. Each `Context` will contain one or more offsets of `JUMPDEST`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ContextJumpDests(pub HashMap<usize, BTreeSet<usize>>);
+
+/// The result after proving a `JumpDestTableWitness`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct JumpDestTableProcessed(HashMap<usize, Vec<usize>>);
+
+/// Map `CodeAddress -> (Context -> [JumpDests])`
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct JumpDestTableWitness(HashMap<H256, ContextJumpDests>);
+
+impl ContextJumpDests {
+    pub fn insert(&mut self, ctx: usize, offset: usize) {
+        self.entry(ctx).or_default().insert(offset);
+    }
+
+    pub fn get(&self, ctx: usize) -> Option<&BTreeSet<usize>> {
+        self.0.get(&ctx)
+    }
+}
+
+impl JumpDestTableProcessed {
+    pub fn new(ctx_map: HashMap<usize, Vec<usize>>) -> Self {
+        Self(ctx_map)
+    }
+}
+
+impl JumpDestTableWitness {
+    pub fn new(ctx_map: HashMap<H256, ContextJumpDests>) -> Self {
```

Comment:
An odd signature - this behaves like a collection, with `get`, `insert` methods, but the constructor breaks the pattern.

Let's do `new` like other collections - if you need this, then implement `From<HashMap<..>>`